freelist.go

```diff
@@ -171,7 +172,9 @@ func (f *freelist) read(p *page) {
 	copy(f.ids, ids)
 
 	// Make sure they're sorted.
-	sort.Sort(pgids(f.ids))
+	if !pgids(f.ids).isSorted() {
```

Comment:
Since you bothered to benchmark, why didn't you include it in the commit? And it's not a question of how much faster it is in isolation, it's a question of how much faster it is in the context of things that matter in boltdb.

Anyway, I'll stop reviewing now.